Makefile

```diff
@@ -275,7 +275,7 @@ build-all-new-commits:
 
 .PHONY: vendor
 vendor:
-	$(GO) mod tidy
+	$(GO) mod tidy -compat=1.17 
```

Comment:
What happens without this change?

Reply:
go changed its go.mod format in newer versions, so if you try to vendor locally with something newer it creates something different from the go 1.17 version in CI thus always causing the diff checker to trip.
Obviously we need to keep making the old go version happy so the compat flag should ensure all go version will use the same format so this change makes sense.